src/psyclone/tests/gocean1p0_stencil_test.py

```diff
@@ -0,0 +1,227 @@
+    stencil_string = "random_name"
+    parsed_stencil = expr.FORT_EXPRESSION.parseString(stencil_string)[0]
+    with pytest.raises(ParseError) as excinfo:
+        stencil.load(parsed_stencil, "kernel_stencil")
+    assert ("argument is 'random_name' but must be one of ['pointwise'] or "
+            "stencil(...)") in str(excinfo.value)
+
+    # this should cause an unsupported name error
+    stencil_string = "stenci(a)"
+    parsed_stencil = expr.FORT_EXPRESSION.parseString(stencil_string)[0]
+    with pytest.raises(ParseError) as excinfo:
+        stencil.load(parsed_stencil, "kernel_stencil")
+    assert "argument is 'stenci' but must be 'stencil(...)" \
+        in str(excinfo.value)
+
+    # this should cause a not-enough-args error
+    stencil_string = "stencil(a)"
+    parsed_stencil = expr.FORT_EXPRESSION.parseString(stencil_string)[0]
+    with pytest.raises(ParseError) as excinfo:
+        stencil.load(parsed_stencil, "kernel_stencil")
+    assert "format 'stencil(...)', has 1 arguments but should have 3" \
+        in str(excinfo.value)
+
+    # this should cause a not-a-number error
+    stencil_string = "stencil(a,b,c)"
+    parsed_stencil = expr.FORT_EXPRESSION.parseString(stencil_string)[0]
+    with pytest.raises(ParseError) as excinfo:
+        stencil.load(parsed_stencil, "kernel_stencil")
+    assert "Argument index 0 should be a number but found 'a'" \
+        in str(excinfo.value)
+
+    # this should also cause a not-a-number error
+    stencil_string = "stencil(000,x00,000)"
+    parsed_stencil = expr.FORT_EXPRESSION.parseString(stencil_string)[0]
+    with pytest.raises(ParseError) as excinfo:
+        stencil.load(parsed_stencil, "kernel_stencil")
+    assert "index 1 should be a number but found 'x00'" \
+        in str(excinfo.value)
+
+    # this should cause a not-3-numbers error
+    stencil_string = "stencil(012,345,67)"
+    parsed_stencil = expr.FORT_EXPRESSION.parseString(stencil_string)[0]
+    with pytest.raises(ParseError) as excinfo:
+        stencil.load(parsed_stencil, "kernel_stencil")
+    assert "index 2 should consist of 3 numbers but found 2" \
+        in str(excinfo.value)
+
+    # this should cause an invalid-middle-number error
+    stencil_string = "stencil(012,345,678)"
+    parsed_stencil = expr.FORT_EXPRESSION.parseString(stencil_string)[0]
+    with pytest.raises(ParseError) as excinfo:
+        stencil.load(parsed_stencil, "kernel_stencil")
+    assert "index 1 position 1 should be a number from 0-1 but found 4" \
+        in str(excinfo.value)
+
+    # this should cause a zero-size-stencil error
+    stencil_string = "stencil(000,010,000)"
+    parsed_stencil = expr.FORT_EXPRESSION.parseString(stencil_string)[0]
+    with pytest.raises(ParseError) as excinfo:
+        stencil.load(parsed_stencil, "kernel_stencil")
+    assert ("A zero sized stencil has been specified. This should be "
+            "specified with the 'pointwise' keyword") in str(excinfo.value)
+
```

Comment:
What about if the user gets confused and tries to do e.g. "stencil(-100,-111,000)"?

Reply:
You get the following error:

Parse Error: Meta-data error in kernel 'kernel_stencil': 3rd descriptor (stencil) of field argument with format 'stencil(...)'. Argument index 0 should consist of 3 digits but found 4.

Is that OK? Do you want me to add it as a test?

Reply:
No, that's OK.